internal/services/recoveryservices/recovery_services_vault_resource_test.go

```diff
@@ -188,13 +188,28 @@ func TestAccRecoveryServicesVault_UserAssignedIdentity(t *testing.T) {
 	})
 }
 
+func TestAccRecoveryServicesVault_immutabilityLocked(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_recovery_services_vault", "test")
+	r := RecoveryServicesVaultResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			// To test creation with `Locked`, it is irreversible.
```

Comment:
Since this is irreversible, shall we also test/implement that when this property get updated, the resource will be recreated?